src/Npgsql/PoolManager.cs

```diff
@@ -71,7 +72,7 @@ internal static ConnectorPool GetOrAdd(string key, ConnectorPool pool)
                 {
                     var newPools = new (string, ConnectorPool)[_pools.Length * 2];
                     Array.Copy(_pools, newPools, _pools.Length);
-                    Interlocked.Exchange(ref _pools, newPools);
+                    _pools = newPools;
```

Comment:
I'm not good enough with low-level memory architecture, but I seem to remember that `Interlocked.Exchange()` provides a stronger guarantee than simply a volatile variable (that's the main reason I put on here in the first place). Specifically I think it guarantees the new value is seen on other cores immediately (cache flush), and not just preventing of reordering.

@YohDeadfall if you think this is wrong I can try to look where I got that from. Otherwise it may be safer just to leave the `Interlocked.Exchange()` - this is a super-rare event anyway so we don't care so much.

Reply:
> I'm not good enough with low-level memory architecture

Me too, but I read articles to verify my assumptions and knowledge. Here they are:

* [C# - The C# Memory Model in Theory and Practice](https://msdn.microsoft.com/en-us/magazine/jj863136.aspx)
* [C# - The C# Memory Model in Theory and Practice, Part 2](https://msdn.microsoft.com/en-us/magazine/jj883956.aspx)
* [Volatile keyword in C# – memory model explained](http://igoro.com/archive/volatile-keyword-in-c-memory-model-explained/)

So everything goes fine, no reasons to worry about (:

Reply:
OK for me!